Missing `#include <limits>`

When compiling MIOpen with OpenCL backend (release 4.3.0) on Arch Linux with GCC 11 I get several build errors like
```C++
MIOpen/src/include/miopen/float_equal.hpp:47:39: error: no member named 'numeric_limits' in namespace 'std'
               std::nextafter(x, std::numeric_limits<T>::lowest()) <= y
```
This is fixed by adding `#include <limits>` to `float_equal.hpp`:

```patch
diff --git a/src/include/miopen/float_equal.hpp b/src/include/miopen/float_equal.hpp
index e2441b60c..bd816cb22 100644
--- a/src/include/miopen/float_equal.hpp
+++ b/src/include/miopen/float_equal.hpp
@@ -28,6 +28,7 @@
 
 #include <algorithm>
 #include <cmath>
+#include <limits>
 #include <numeric>
 #ifdef _MSC_VER
 #include <iso646.h>

```

--- a/src/include/miopen/float_equal.hpp
+++ b/src/include/miopen/float_equal.hpp
@@ -28,6 +28,7 @@
 
 #include <algorithm>
 #include <cmath>
+#include <limits>
 #include <numeric>
 #ifdef _MSC_VER
 #include <iso646.h>

